### PR TITLE
jump-heading: clean headings, specially for Wikipedia

### PR DESCRIPTION
**Pull request by vindarel**
Wikipedia headings end with "[edit]", in french it's an even longer
"[modifier | modifier le code]". That's ugly.
We strip what comes after the first bracket.

for #155

We need a list of real candidates and a list of user-friendly names, hence the elegant use of a closure. I found no built-in mechanism for this in `read-from-minibuffer`…

### CODE REVIEW

File: source/jump-heading.lisp [hunk 1]
```diff
@@ -20,15 +20,36 @@
...
-                           :input-prompt "Jump to heading:"
-                           :completion-function (lambda (input)
-                                                  (fuzzy-match
-                                                   input
-                                                   (cl-json:decode-json-from-string
-                                                    headings))))))
-    (paren-jump-to-heading :heading-inner-text heading)))
+(defun clean-heading (heading)
+  "Cleanup a heading.
+For example, wikipedia ones end with '[edit]'. We strip what comes after the first bracket."
```
Comment (Ambrevar) [inline source/jump-heading.lisp]: Wikipedia (capital).

File: source/jump-heading.lisp [hunk 2]
```diff
@@ -20,15 +20,36 @@
...
+      (cleaned-headings nil))
+  ;; We need to preserve the list of real headings, the ones that are the names of links,
+  ;; but we want to present cleaner names to the user in the completion buffer.
+  (defun clean-headings (headings/json)
+    "Transform the json string to a list of headings, and clean them up."
+    (check-type headings/json string)
+    (let ((headings (cl-json:decode-json-from-string headings/json)))
+      (setf current-headings headings)
+      (setf cleaned-headings (mapcar #'clean-heading headings))
+      cleaned-headings))
```
Comment (Ambrevar) [inline source/jump-heading.lisp]: The current way we use to general display value from real values is with the `object-string` method.  So maybe it's easier to just create a `heading` class or type and implement the method.
Thoughts?

File: source/jump-heading.lisp [hunk 3]
```diff
@@ -20,15 +20,36 @@
...
+
+  (defun heading-from-selection (selection)
+    "From the user selection, which is a heading cleaned up, get back the original heading string, so than we can jump to it."
```
Comment (Ambrevar) [inline source/jump-heading.lisp]: Line is too long.

File: source/jump-heading.lisp [hunk 4]
```diff
@@ -20,15 +20,36 @@
...
+    (elt current-headings
+         (position selection cleaned-headings :test #'string=)))
+
+  (define-command jump-to-heading ()
+    "Jump to a particular heading, of type h1, h2, h3, h4, h5, or h6"
```
Comment (Ambrevar) [inline source/jump-heading.lisp]: Period missing.

**Comment by vindarel** [comment]
It's much better indeed with object-string.

**Pull request merged**